LSP error when stopping OH

Running OH snapshot 1104, I got this error when stopping OH:
```
2017-12-02 09:21:24.729 [ERROR] [thome.model.lsp.internal.ModelServer] - Error accepting the client connection
java.net.SocketException: Socket closed
        at java.net.PlainSocketImpl.socketAccept(Native Method) ~[?:?]
        at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409) ~[?:?]
        at java.net.ServerSocket.implAccept(ServerSocket.java:545) ~[?:?]
        at java.net.ServerSocket.accept(ServerSocket.java:513) ~[?:?]
        at org.eclipse.smarthome.model.lsp.internal.ModelServer.listen(ModelServer.java:80) ~[?:?]
        at org.eclipse.smarthome.model.lsp.internal.ModelServer.lambda$0(ModelServer.java:58) ~[?:?]
        at java.lang.Thread.run(Thread.java:745) [?:?]
```

--- a/bundles/model/org.eclipse.smarthome.model.lsp/src/main/java/org/eclipse/smarthome/model/lsp/internal/ModelServer.java
+++ b/bundles/model/org.eclipse.smarthome.model.lsp/src/main/java/org/eclipse/smarthome/model/lsp/internal/ModelServer.java
@@ -51,11 +51,8 @@ public class ModelServer {
     private ScriptEngine scriptEngine;
     private Injector injector;
 
-    private boolean isShuttingDown = false;
-
     @Activate
     public void activate() {
-        isShuttingDown = false;
         injector = Guice.createInjector(new RuntimeServerModule(scriptServiceUtil, scriptEngine));
         new Thread(() -> {
             listen();
@@ -64,7 +61,6 @@ public class ModelServer {
 
     @Deactivate
     public void deactivate() {
-        isShuttingDown = true;
         try {
             if (socket != null && !socket.isClosed()) {
                 socket.close();
@@ -86,8 +82,8 @@ public class ModelServer {
                         handleConnection(client);
                     }, "Client " + client.getRemoteSocketAddress()).start();
                 } catch (IOException e) {
-                    if (!isShuttingDown) {
-                        logger.error("Error opening socket for client connections: {}", e.getMessage());
+                    if (!socket.isClosed()) {
+                        logger.error("Error accepting client connection: {}", e.getMessage());
                     }
                 }
             }

--- a/bundles/model/org.eclipse.smarthome.model.lsp/src/main/java/org/eclipse/smarthome/model/lsp/internal/ModelServer.java
+++ b/bundles/model/org.eclipse.smarthome.model.lsp/src/main/java/org/eclipse/smarthome/model/lsp/internal/ModelServer.java
@@ -51,8 +51,11 @@ public class ModelServer {
     private ScriptEngine scriptEngine;
     private Injector injector;
 
+    private boolean isShuttingDown = false;
+
     @Activate
     public void activate() {
+        isShuttingDown = false;
         injector = Guice.createInjector(new RuntimeServerModule(scriptServiceUtil, scriptEngine));
         new Thread(() -> {
             listen();
@@ -61,6 +64,7 @@ public class ModelServer {
 
     @Deactivate
     public void deactivate() {
+        isShuttingDown = true;
         try {
             if (socket != null && !socket.isClosed()) {
                 socket.close();
@@ -82,7 +86,9 @@ public class ModelServer {
                         handleConnection(client);
                     }, "Client " + client.getRemoteSocketAddress()).start();
                 } catch (IOException e) {
-                    logger.debug("Error accepting the client connection: {}", e.getMessage());
+                    if (!isShuttingDown) {
+                        logger.error("Error opening socket for client connections: {}", e.getMessage());
+                    }
                 }
             }
         } catch (IOException e) {

--- a/bundles/model/org.eclipse.smarthome.model.lsp/src/main/java/org/eclipse/smarthome/model/lsp/internal/ModelServer.java
+++ b/bundles/model/org.eclipse.smarthome.model.lsp/src/main/java/org/eclipse/smarthome/model/lsp/internal/ModelServer.java
@@ -82,7 +82,7 @@ public class ModelServer {
                         handleConnection(client);
                     }, "Client " + client.getRemoteSocketAddress()).start();
                 } catch (IOException e) {
-                    logger.error("Error accepting the client connection", e);
+                    logger.debug("Error accepting the client connection: {}", e.getMessage());
                 }
             }
         } catch (IOException e) {

